tsconfig.json

```diff
@@ -25,7 +25,7 @@
     // "isolatedModules": true,               /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
 
     /* Strict Type-Checking Options */
-    "strict": false,                           /* Enable all strict type-checking options. */
+    "strict": true,                           /* Enable all strict type-checking options. */
```

Comment:
I should have reviewed this file first :) I would say that it might be wise for sanity purposes to turn these checks on one-at-a-time so that the PRs are smaller. As you can see I made a lot of comments about all sorts of these different things. I would like a second set of eyes on a PR this large, and, given how much we've broken others peoples code with our TypeScript changes, it seems prudent to bite off smaller chunks at a time.

I'm all for starting our 2.0 release cycle and not publishing until we are done with all these changes. If there is a bug-fix or security thing that needs to be addressed, I think we can branch from the current 1.5 tag and fix it there and then merge that back in to `master`.

Thoughts?